icefoganalytics/elcc-data-management
# ELCC-28: Support Cusomtization of Wage Enhancements Section

Fixes https://yg-hpw.atlassian.net/browse/ELCC-28

Relates to:

- [2022-23 Child Care Centres E-H -> Wage Enhancement](https://icefoganalyticscom.sharepoint.com/:x:/s/Project_Files/EXZzX-POQVxCiG_i7GLUBBIBGZEHHVaipJeyFUZCUx0ciA?e=DvdEnh&nav=MTJfQTEwMl97MDAwMDAwMDAtMDAwMS0wMDAwLTAyMDAtMDAwMDAwMDAwMDAwfQ)

# Context

Wage enhancement section needs more spaces – and the ability to add additional – forgive me if it does this and I just can’t figure it out…

Any given daycare or facility may have multiple personnel at each level of certification (1,2,3).  Currently the system only allows for entry of one employee of each type.

Can the system dynamically allocate additional rows for each level as required by the officer entering the information?   We don’t want 5 rows for each level for instance as not all of them will be used on a regular basis.  The ability to add more rows as needed per level is what is required.

Requirements

pull “Wage enhancements” section out of the “worksheets” tab, and built it its own modeling on and widget on the “Employees” tab.

Current Implementation

![image](https://github.com/icefoganalytics/elcc-data-management/assets/23045206/d2da0098-0465-4006-8fa8-d889317a6e2c)

Expected Implementation (more or less)
![image](https://github.com/icefoganalytics/elcc-data-management/assets/23045206/43a5aacf-1cba-48a5-b3a4-d7fa2a279a63)

# Implementation

Add back-end modeling and set up some minimal UI component paths.
Add seeders for production for models that don't have an edit interface.

### `employee_wage_tiers` Table

- **Columns**:
  - `id`: Auto-incremented unique ID.
  - `fiscal_period_id`: Links to `fiscal_periods`.
  - `tier_level`: Numeric level of wage tier.
  - `tier_label`: Text label (50 chars max).
  - `wage_rate_per_hour`: Hourly wage rate.
  - `created_at`/`updated_at`: Timestamps.

- **Constraints**:
  - Primary Key: `id`.
  - Foreign Key: `fiscal_period_id`.

- **Index**:
  - Unique on `fiscal_period_id` and `tier_level`.

### `wage_enhancements` Table

- **Columns**:
  - `id`: Auto-incremented unique ID.
  - `centre_id`: Links to `centres`.
  - `employee_wage_tier_id`: Links to `employee_wage_tiers`.
  - `employee_name`: Name of the employee (100 chars max).
  - `hours_estimated`: Estimated work hours.
  - `hours_actual`: Actual work hours.
  - `created_at`/`updated_at`: Timestamps.

- **Constraints**:
  - Primary Key: `id`.
  - Foreign Keys: 
    - `centre_id` to `centres`.
    - `employee_wage_tier_id` to `employee_wage_tiers`.

- **Index**:
  - Unique on `centre_id` and `employee_wage_tier_id`.

# Screenshots

# Testing Instructions

1. Boot the app via `dev up`
2. Check that the app complies, and that you can log in at http://localhost:8080.
3. Go to http://localhost:8080/child-care-centres/1/2023-24/employees/april.
4. Scroll to the bottom of the page and you will see a Wage Enhancements section that renders the different levels.
> UI has not been implemented more than enough to test the modeling.
